addons/website_slides_survey/static/src/js/public/components/slide_upload_dialog/slide_upload_category.js

```diff
@@ -0,0 +1,78 @@
+/** @odoo-module **/
+
+import { SlideUploadCategory } from "@website_slides/js/public/components/slide_upload_dialog/slide_upload_category";
+import { patch } from "@web/core/utils/patch";
+import { uniqueId } from "@web/core/utils/functions";
+import { onWillStart } from "@odoo/owl";
+
+patch(SlideUploadCategory.prototype, {
+    /**
+     * Overridden to add the "certification" slide category
+     */
+    setup() {
+        super.setup();
+        this.state.choices.certifications = [];
+        this.state.choices.certificationId = "";
+        onWillStart(async () => {
+            const certifications = await this._fetch_certifications();
+            this.state.choices.certifications = certifications;
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    onCertificationSelect(value) {
+        this.state.choices.certificationId = value;
+        // Slide title = selected certification's name
+        this.state.slideName = this.state.choices.certifications.find(
+            (c) => c.value === value
+        ).label;
+    },
+
+    onClickCreateCertificationBtn(categoryName) {
+        const tempId = uniqueId("temp");
+        this.state.choices.certifications.push({ value: tempId, label: categoryName });
+        this.state.choices.certificationId = tempId;
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    async _fetch_certifications() {
```

Comment:
Shall we use it after the setup? If not, could be "inlined".

Reply:
I mean not create a private method but use these few lines in the `onWillStart` hook directly.

Reply:
Sure. Let's do it :+1: